apps/gateway/src/middlewares/oai-errors.ts

```diff
@@ -56,6 +60,28 @@ export const oaiErrors = new Elysia({ name: "oai-error" })
       });
     }
 
+    if (error instanceof BadRequestError) {
+      return status(400, {
+        error: {
+          message: error.message,
+          type: error.type,
+          param: undefined,
+          code: error.code,
+        },
+      });
+    }
+
+    if (error instanceof UpstreamAuthFailedError) {
```

Comment:
Why don't we just construct the error shape in the way that it is returned in the first place?